src/components/RightSideDrawer.jsx

```diff
@@ -0,0 +1,110 @@
+import React from "react";
+
+import Brightness4Icon from "@mui/icons-material/Brightness4";
+import Brightness7Icon from "@mui/icons-material/Brightness7";
+import LanguageIcon from "@mui/icons-material/Language";
+import MenuIcon from "@mui/icons-material/Menu";
+import PaletteIcon from "@mui/icons-material/Palette";
+import SettingsIcon from "@mui/icons-material/Settings";
+import { Box, List, ListItem, ListItemIcon, ListItemText, Divider, Select, MenuItem } from "@mui/material";
+import Avatar from "@mui/material/Avatar";
+import Drawer from "@mui/material/Drawer";
+import IconButton from "@mui/material/IconButton";
+import PropTypes from "prop-types";
+
+const RightSideDrawer = ({ themeMode, toggleThemeMode }) => {
+  const [open, setOpen] = React.useState(false);
+  const [isLoggedIn, setIsLoggedIn] = React.useState(false); // Assuming user login state
+  const [selectedColor, setSelectedColor] = React.useState("primary"); // 'primary' or 'secondary'
+  const [language, setLanguage] = React.useState("English"); // Default language
+  const languageList = ["English", "Spanish", "French", "German", "Italian"]; // Example language list
+
+  const handleToggleDrawer = () => {
+    setOpen(!open);
+  };
+
+  const handleColorChange = (event) => {
```

Comment:
use callback  to function that passed to child component as prop , to prevent   re-rendering